test/dynamo/test_functions.py

```diff
@@ -330,6 +332,58 @@ def test_methodcall2(a, b):
     def test_methodcall3(a, b):
         return constant3(a, b=1.0) + b
 
+    def test_is_integer(self):
+        @torch._dynamo.optimize(nopython=True)
+        def forward(t, m):
+            return t if m.is_integer() else 2 * t
+
+        t = torch.tensor([1])
+        assert forward(t, 1.0).item() == 1
+        assert forward(t, 1.5).item() == 2
+
+        assert forward(t, np.float32(1.0)).item() == 1
+        assert forward(t, np.float64(1.5)).item() == 2
+
+    @parametrize("method", ("as_integer_ratio", "bit_length", "conjugate"))
+    def test_int_methods(self, method):
+        @torch._dynamo.optimize(nopython=True)
+        def forward(t, m):
+            return t if getattr(m, method)() else 2 * t
+
+        t = torch.tensor([1])
+        forward(t, 1)
+
+    @parametrize("method", ("as_integer_ratio", "conjugate", "hex", "is_integer"))
+    def test_float_methods(self, method):
+        @torch._dynamo.optimize(nopython=True)
+        def forward(t, m):
+            return t if getattr(m, method)() else 2 * t
+
+        t = torch.tensor([1])
+        forward(t, 1.0)
+
+    def test_fraction_as_integer_ratio(self):
+        @torch._dynamo.optimize(nopython=True)
+        def forward(t, m):
+            return t if m.as_integer_ratio()[1] else 2 * t
+
+        t = torch.tensor([1])
+        forward(t, fractions.Fraction(1))
+
+    @pytest.mark.xfail(
+        reason=(
+            "Fraction.conjugate() raises torch._dynamo.exc.Unsupported: "
+            "call_function UserDefinedClassVariable"
+        )
+    )
+    def test_fraction_conjugate(self):
+        @torch._dynamo.optimize(nopython=True)
+        def forward(t, m):
+            return t if m.conjugate() else 2 * t
+
+        t = torch.tensor([1])
+        forward(t, fractions.Fraction(1))
```

Comment:
Fine if we can support thests fractions for free, but I wouldnt' worry too much about them if we don't.

Reply:
Well, we can support fractions for free, _but_ given that we can't support numpy scalars (yet), I changed the condition back to `isinstance(..., (float, int))`

I felt that no one was actually going to use `fractions` in serious work: I tested it only because I was using the more general `numbers.Number` as the test, and that's the only other type that implements `numbers.Number` in the standard library, though now I say this I'm not 100% sure there isn't some other I haven't thought of.

Reply:
We can support NumPy scalars, only that the support is not to be implemented here, but in the whole `np.array` class. It's fine if we create a new method for generic `np.arrays` (other than 0D) that they don't really have in NumPy.

Reply:
In this case, you'll need to implement the method yourself, rather than being able to forward it to an object, given that we don't have a NumPy scalar backing the np.array.